<launch>
  <arg name="buffer_size" default="3" doc="The number of entity messages to buffer to use for smoothing/extrapolation.  Must be a positive integer." />
  <arg name="headless" default="true" doc="Controls whether SMARTS should also emit its state to an Envision server." />
  <arg name="seed" default="42" doc="Seed to use when initializing SMARTS' random number generator(s)." />
  <arg name="target_freq" default="0" doc="The target frequencey in Hz.  If not specified, the node will publish as quickly as SMARTS permits." />

  <arg name="test_scenario" default="scenarios/loop" doc="A path to scenario to test with." />
  <arg name="test_agent_type" default="module:agent" doc="A zoo registry locator with which to find the test agent." />
  <arg name="test_agent_params" default="" doc="Optional JSON string containing parameters to configure the test agent." />
  <arg name="test_agent_start_pose" default="[[0.0, 0.0, 0.0], 0.0]" doc="The starting pose for the test agent." />
  <arg name="test_agent_speed" default="0" doc="Optional starting speed for the test agent." />

  <node pkg="smarts_ros" name="SMARTS" type="ros_driver.py" output="screen">
    <param name="_target_freq" type="double" value="$(arg target_freq)" />
    <param name="_buffer_size" type="int" value="$(arg buffer_size)" />
    <param name="_headless" type="bool" value="$(arg headless)" />
    <param name="_seed" type="int" value="$(arg seed)" />
  </node>

  <test test-name="test_smarts_ros" pkg="smarts_ros" type="test_smarts_ros.py">
    <param name="scenario" type="str" value="$(arg test_scenario)" />
    <param name="agent_type" type="str" value="$(arg test_agent_type)" />
    <param name="agent_params" type="str" value="$(arg test_agent_params)" />
    <param name="agent_start_pose" type="str" value="$(arg test_agent_start_pose)" />
    <param name="agent_speed" type="double" value="$(arg test_agent_speed)" />
  </test>
</launch>
